### ISSUE TITLE
Provide an explicit name for the built image

### ISSUE THREAD

**Issue by arun-gupta** [docker/compose#2498]
Building an image using the following Compose file:

```
mycouchbase:
  build: .
  ports:
    - 8091:8091
    - 8092:8092 
    - 8093:8093 
    - 11210:11210
```

The generated image name is `couchbase_mycouchbase`. There should be an `image_name` key that will allow to override the default image name. The updated syntax can be:

```
mycouchbase:
  build: .
  image_name: arungupta/couchbase
  ports:
    - 8091:8091
    - 8092:8092 
    - 8093:8093 
    - 11210:11210
```

#541 addresses a different concern.

**Comment by dnephin** [comment]
This will be addressed by #2092. The `image` field will be used as the image name.

**Comment by arun-gupta** [comment]
@dnephin thanks, what are the timelines for 1.6.0?
